runtime-config.md

```diff
@@ -22,6 +23,13 @@ In Linux, for e.g., the network namespace could be configured in this hook.
 
 If a hook returns a non-zero exit code, then an error including the exit code and the stderr is returned to the caller and the container is torn down.
 
+### Post-start
+
+The post-start hooks are called after the user process is started.
+It can be used for notifying user that real process is spawned.
```

Comment:
On Tue, Sep 29, 2015 at 01:33:06PM -0700, Alexander Morozov wrote:
> Adding to consul or to some other discovery tool…

Hmm, wouldn't this need the service to be listening on a socket?
Registering it in a post-start hook would be better than registering
it in a pre-start hook, but it would still be too early ;).

> Apart from this I have real use-case in docker where I need event
> when I can say client that application is ready.

Again, “is ready” is likely to be some time after “we just exec-ed
it”.

> Do you really have plans to start `sidecar` processes?

@julz is working on garden for cloudfoundary [2], and they have
explicit create-container [3] and launch-application [4] steps.  He
needs a way to know when the container is safe to exec into so he can
allow/deny launch-application requests.  See the “How do we know when
the process is up?” section of [5].

That's not a sidecar container, but it is close to a sidecar
application.  And “sidecar” is familiar enough language that I think
it gets the point across without a long-winded explanation ;).

[1]: https://github.com/opencontainers/specs/wiki/MeetingMinutes:-2015-09-16
[2]: https://github.com/julz/garden-runc
[3]: https://github.com/cloudfoundry-incubator/garden/blob/bf9d38d47411f79be123005411040568217ef874/doc/garden-api.md#create-a-new-container
[4]: https://github.com/cloudfoundry-incubator/garden/blob/bf9d38d47411f79be123005411040568217ef874/doc/garden-api.md#run-a-process-inside-a-container
[5]: https://github.com/opencontainers/specs/wiki/MeetingMinutes:-2015-09-16